zebra-state/src/in_memory.rs

```diff
@@ -0,0 +1,119 @@
+use super::{Request, Response};
+use futures::prelude::*;
+use std::{
+    collections::{BTreeMap, HashMap},
+    error::Error,
+    future::Future,
+    pin::Pin,
+    sync::Arc,
+    task::{Context, Poll},
+};
+use tower::{buffer::Buffer, Service};
+use zebra_chain::{
+    block::{Block, BlockHeaderHash},
+    types::BlockHeight,
+};
+
+#[derive(Default)]
+struct ZebraState {
+    by_hash: HashMap<BlockHeaderHash, Arc<Block>>,
+    by_height: BTreeMap<BlockHeight, Arc<Block>>,
+}
+
+impl ZebraState {
+    fn insert(&mut self, block: impl Into<Arc<Block>>) {
```

Comment:
What impls `Into<Arc<Block>>`?

Reply:
Block and Arc<Block> and probably Box<Block> and who knows what other rust smart pointers

Reply:
ah cool! thanks for the tip